### PR TITLE
feat: Add getInitialSessionAuthContext to compute SessionAuth context

### PR DESCRIPTION
**Pull request by sasha240100**
## Summary of change

Adds `getInitialSessionAuthContext` to support computing initial `SessionAuth` context (to be used with `initialSessionAuthContext` property from  https://github.com/supertokens/supertokens-auth-react/pull/789)

## Related issues

-   https://github.com/supertokens/supertokens-auth-react/issues/790
-   https://github.com/supertokens/supertokens-auth-react/pull/789

## Test Plan

Tested locally on a NextJS implementation using supertokens. `getInitialSessionAuthContext(session)` creates an SSR session context that works like expected when passed into `initialSessionAuthContext={}` of `<SessionAuth />` component. Also covered by unit tests

## Documentation changes

TODO.

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `lib/ts/version.ts`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `lib/ts/version.ts`
-   [x] Had run `npm run build-pretty`
-   [x] Had installed and ran the pre-commit hook
-   [ ] If new thirdparty provider is added,
    -   [ ] update switch statement in `recipe/thirdparty/providers/configUtils.ts` file, `createProvider` function.
    -   [ ] add an icon on the user management dashboard.
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `add-ts-no-check.js` file to include that
-   [ ] If added a new recipe / api interface, then make sure that the implementation of it uses NON arrow functions only (like `someFunc: function () {..}`).
-   [ ] If added a new recipe, then make sure to expose it inside the recipe folder present in the root of this repo. We also need to expose its types.

## Remaining TODOs for this PR

-   [ ] Add to documentation

### CODE REVIEW

File: lib/ts/types.ts [hunk 1]
```diff
@@ -113,3 +114,13 @@ export type User = {
     // the recipeUserId can be converted to string from the RecipeUserId object type.
     toJson: () => any;
 };
+
+export type SSRSessionContextType = {
+    isContextFromSSR: true;
+    loading: false;
+    doesSessionExist: boolean;
+    accessTokenPayload: any;
+    invalidClaims: ClaimValidationError[]; // TODO: Find if it's possible to compute invalidClaims on BE
```
Comment (porcellus) [inline lib/ts/types.ts:123]: As mentioned in the auth-react PR, if we can't find a way to do these properly (which I don't think we can, not right now at least), the claim validation related props should be removed from this type.

File: lib/ts/types.ts [hunk 2]
```diff
@@ -113,3 +114,13 @@ export type User = {
     // the recipeUserId can be converted to string from the RecipeUserId object type.
     toJson: () => any;
 };
+
+export type SSRSessionContextType = {
+    isContextFromSSR: true;
+    loading: false;
```
Comment (porcellus) [inline lib/ts/types.ts:120]: These should not be required since they are static values.

**Review by porcellus — CHANGES_REQUESTED** [review]

File: lib/ts/types.ts (hunk 2 above)
Comment (sasha240100) [reply lib/ts/types.ts:119]: But they are expected by `supertokens-auth-react`, if they will be optional - that means we would have to polyfill them inside `SessionAuth` component, and in that case I would rather remove them completely from `supertokens-node`

File: lib/ts/nextjs.ts [hunk 3]
```diff
@@ -230,6 +230,24 @@ export default class NextJS {
         return result;
     }
 
+    static async getInitialSessionAuthContext(session: SessionContainer | undefined): Promise<SSRSessionContextType> {
```
Comment (rishabhpoddar) [inline lib/ts/nextjs.ts:233]: I don't think we need this function. And if we are going to keep it, why is it in the nextjs file?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

**Pull request closed without merging**